blocks/button-pfx/button-pfx.js

```diff
@@ -0,0 +1,52 @@
+export default async function decorate(block) {
+  const [link, buttonLabel, type] = block.children;
+  block.textContent = '';
+
+  if (link.textContent.trim() === '') {
+    const buttonEl = document.createElement('button');
+    buttonEl.classList.add('button');
+    buttonEl.textContent = buttonLabel.textContent.trim();
+    block.appendChild(buttonEl);
+  } else {
+    const buttonEl = document.createElement('a');
+    buttonEl.classList.add('button');
+    buttonEl.textContent = buttonLabel.textContent.trim();
+    buttonEl.href = link.textContent.trim();
+    block.appendChild(buttonEl);
+  }
+
+  if (type.textContent.trim() === 'primary-dropdown') {
```

Comment:
Considering performance can you please convert this if else case to a Switch case?